### PR TITLE
:seedling: Fix E2E log collector regression by aligning to the container naming rules used in CAPD

### PR DESCRIPTION
**Pull request by fabriziopandini**
Fixes a regression in theE2E log collector error introduced by https://github.com/kubernetes-sigs/cluster-api/pull/3600 😅 

`Failed to get logs for machine quick-start-6fnlyv-md-0-7b8b4fc4d5-hl89q, cluster quick-start-e7nyih/quick-start-6fnlyv: [command "docker exec --privileged quick-start-6fnlyv-md-0-7b8b4fc4d5-hl89q journalctl --no-pager --output=short-precise -u kubelet.service" failed with error: exit status 1, command "docker exec --privileged quick-start-6fnlyv-md-0-7b8b4fc4d5-hl89q journalctl --no-pager --output=short-precise -u containerd.service" failed with error: exit status 1, command "docker exec --privileged quick-start-6fnlyv-md-0-7b8b4fc4d5-hl89q journalctl --no-pager --output=short-precise" failed with error: exit status 1, command "docker exec --privileged quick-start-6fnlyv-md-0-7b8b4fc4d5-hl89q kubelet --version" failed with error: exit status 1, command "docker exec --privileged quick-start-6fnlyv-md-0-7b8b4fc4d5-hl89q journalctl --no-pager --output=short-precise -k" failed with error: exit status 1, command "docker exec --privileged quick-start-6fnlyv-md-0-7b8b4fc4d5-hl89q crictl info" failed with error: exit status 1]`

### CODE REVIEW

File: test/framework/docker_logcollector.go [hunk 1]
```diff
@@ -31,8 +32,15 @@ import (
 // DockerLogCollector collect logs from a CAPD workload cluster.
 type DockerLogCollector struct{}
 
+func machineContainerName(cluster, machine string) string {
+	if strings.HasPrefix(machine, cluster) {
+		return machine
+	}
+	return fmt.Sprintf("%s-%s", cluster, machine)
+}
```
Comment (vincepri) [inline test/framework/docker_logcollector.go:40]: Can we add a comment on why is this needed? And in the PR title as well please

Comment (fabriziopandini) [reply test/framework/docker_logcollector.go:43]: done!

**Comment by fabriziopandini** [comment]
/retitle Fix E2E log collector regression by aligning to the container naming rules used in CAPD

**Review by vincepri — APPROVED** [review]
/approve
/lgtm
